src/exoticatechnologies/util/CircleUtils.kt

```diff
@@ -0,0 +1,820 @@
+                                particleSize = particleSize,
+                                particleDuration = particleDuration,
+                                particleColors = particleColors,
+                                particleDrawMode = particleDrawMode
+                            )
+                            // deduct points from that arm
+                            particlePoints[armIndex].removePoints(particlesToDrawPerInterval)
+                        }
+
+                        // now, decrement lastDrawnArm for next iteration
+                        lastDrawnArm = wrapAroundMod(lastDrawnArm - 1, particlePoints.size)
+                        // And bump the drawIterations and previousArmPostDrawIndex
+                        drawIterations++
+                        previousArmPostDrawIndex = particlePoints[lastDrawnArm].getCurrentPointIndex()
+
+                        // And return nothing
+                        Unit
+                    }
+                }.exhaustive
+            }
+        }
+
+        /**
+         * Method for checking whether all of this [Swirl]'s particle arms ([SwirlArmParticles] have finished or not
+         *
+         * @return whether all particle arms have finished or not
+         */
+        fun hasFinished(): Boolean {
+            return particlePoints.all { it.hasFinished() }
+        }
+
+        inner class SwirlArmParticles(
+            private val particlePoints: MutableList<Vector2f> = mutableListOf()
+        ) {
+            private var armSize: Int = 0
+            private var isFinished = false
+
+            /**
+             * Draws the arm depending on the [particleDrawMode]
+             *
+             * For [ParticleDrawMode.ONE_AT_A_TIME], it draws the first point in the list and then removes it from the list of points
+             * For [ParticleDrawMode.WHOLE_ARM], it draws the whole arm and **DOESN'T** remove any points from it.
+             * **NOTE:** In this mode, the points have to be externally removed, because otherwise the arm will **NEVER** finish.
+             *
+             * @param particleSize the particle size to use. Defaults to 12f.
+             * @param particleDuration the lifetime of a single particle, in seconds. Defaults to 0.25f
+             * @param particleColors depending on [particleDrawMode], it should either be a one-element list for [ParticleDrawMode.ONE_AT_A_TIME]
+             * or a list of colors for [ParticleDrawMode.WHOLE_ARM]
+             * @param particleBrightness particle brightness to use. Defaults to 1f
+             * @param particleDrawMode the mode in which to draw particles. See [ParticleDrawMode]
+             */
+            fun draw(
+                particleSize: Float = 12f,
+                particleDuration: Float = 0.25f,
+                particleColors: List<Color>,
+                particleBrightness: Float = 1f,
+                particleDrawMode: ParticleDrawMode
+            ) {
```

Comment:
would be great if this could be further parametrized to **not** remove particles from the list so that we can actually have a contiously-drawing swirl going on for the e.g. missile magnet system...

something along the lines of 
```
removeParticleAfterDrawing: Boolean = true
```

Reply:
i mean, i know - the whole thing is designed around draining all points and then considering itself done.
i'm just saying - this thing **should** be able to support a continuously-drawing swirl as well. an infinite swirl.